### PR TITLE
Add JSONNET_PATH environment variable.

### LINKED ISSUE #411: FR: Allow configuring library search path from environment variable
It would be more convenient to set jsonnet library search path from environment variable instead of from argument.

```
export JSONNETPATH="${HOME}/abc"
jsonnet "$@"
# ==> equivalent to
# jsonnet -J "${JSONNETPATH}" "$@"
```

### PR DESCRIPTION
**Pull request by sparkprime**
Fix #411 (for cpp only)

### CODE REVIEW

**Comment by sparkprime** [comment]
I didn't factor it out but I did make it about half as complex so I think it's fine now.

File: cmd/jsonnet.cpp [hunk 1]
```diff
@@ -707,17 +708,11 @@ int main(int argc, const char **argv)
...
-                const char *path = strtok(input, const_cast<char*>(PATH_SEP));
-                input = nullptr;
-                if (path != nullptr) {
-                    jpath.push_front(path);
-                } else {
-                    break;
-                }
-            } while (true);
-            free(input);
+            std::stringstream ss(jsonnet_path_env);
```
Comment (marcelocantos) [inline cmd/jsonnet.cpp]: Use `std::ostringstream oss(…);` instead.

File: cmd/jsonnet.cpp [hunk 2]
```diff
@@ -707,17 +708,11 @@ int main(int argc, const char **argv)
...
+            std::string path;
+            while (std::getline(ss, path, PATH_SEP)) {
+                jpath.push_front(path);
+            }
             for (const std::string &path : jpath) {
```
Comment (marcelocantos) [inline cmd/jsonnet.cpp]: Fold the two loops into one for good measure.

File: cmd/jsonnet.cpp [hunk 3]
```diff
@@ -707,17 +708,11 @@ int main(int argc, const char **argv)
         const char *jsonnet_path_env = getenv("JSONNET_PATH");
         if (jsonnet_path_env != nullptr) {
```
Comment (marcelocantos) [inline cmd/jsonnet.cpp]: Suggest: `if (const char *jsonnet_path_env = getenv("JSONNET_PATH") {`

File: cmd/jsonnet.cpp (hunk 2 above)
Comment (sparkprime) [reply cmd/jsonnet.cpp]: It's like that to reverse the order.

File: cmd/jsonnet.cpp (hunk 1 above)
Comment (sparkprime) [reply cmd/jsonnet.cpp]: actually `istringstream` but yes, good point :)

**Pull request merged**